lib/spack/spack/compiler.py

```diff
@@ -229,6 +236,9 @@ class Compiler:
     # by any compiler
     _all_compiler_rpath_libraries = ["libc", "libc++", "libstdc++"]
 
+    # string matcher for string representation of platforms supported by a compiler
+    supported_platforms = any
```

Comment:
I think a clearer always-true function would be `lambda x: return True`. `any` expects a list (this works for strings because any string of `>=` 1 character will return true) whereas generally the API for `supported_platforms` is to report, for a specified platform, if it is supported.

Reply:
Ugh, I was actually hoping you'd have a suggestion that avoided the anonymous functions altogether (`any` is also wrong here, you're right about that for sure), I couldn't come up with one in the short period of time I thought about it but IMO an anonymous function pointer as a class attribute is fairly clunky. I'll think about it more today, but if nothing occurs to you (or me) quickly it's fine for now.